docker-compose.test.yml

```
@@ -0,0 +1,45 @@
+---
+# Compose file for test runs
+version: "3.7"
+services:
+  db:
+    image: datarttu/sujuikodb:latest
+    build: .
+    restart: unless-stopped
+    env_file: .env_test
+    volumes:
+      - type: bind
+        source: ./db/db_test
+        target: /docker-entrypoint-initdb.d/
+        read_only: true
+      - type: tmpfs
+        target: /var/lib/postgresql/data
+    healthcheck:
+      test: ["CMD", "pg_isready", "-q", "-U", "${POSTGRES_USER}"]
+      interval: 30s
+      start_period: 5s
+      timeout: 20s
+      retries: 3
+  dataimporter:
+    image: datarttu/sujuikodb:latest
+    depends_on:
+      db:
+        condition: service_healthy
+    restart: on-failure
+    volumes:
+      - type: bind
+        source: ${IMPORT_DATA_DIR}
+        target: /data
+        read_only: true
+    env_file: .env_test
+    # TODO: See if there is a better way to prevent dataimporter from
+    #       trying imports before db is ready.
+    #       Can we utilize healthcheck of db
+    #       like docker-compose < v3 depends_on condition can do?
+    command: ["psql", "-f", "/data/import.sql"]
```

```suggestion
    command: |
      sh -c "
        psql -v ON_ERROR_STOP=on -f /data/import.sql
        status=\"$$?\"
        psql -c \"COPY (SELECT 1) TO PROGRAM 'pg_ctl stop --mode=smart --no-wait';\"
        exit \"$${status}\"
      "
```
This might be a nifty trick towards a runnable test script. After the test run, we close the database (that is why `restart: "no"` for `db`). We return the exit status of the test run which is non-zero on failure due to `ON_ERROR_STOP`.

Note the `$$` and `\"` escapes for `docker-compose`. We also rely on having the authority to close the database server. That is fine in these temporary test runs.

This is not enough to end `docker-compose` with non-zero status in case of `dataimporter` error, but maybe we could use
```sh
docker-compose -f ./docker-compose.test.yml logs dataimporter
```
afterwards in the test script or find the exit code somehow. That can be a separate PR if you wish to do that at all.

---

Cool! I'll try this and we can get back to the test run logic later if needed.